test/e2e/maxconn_test.go

```diff
@@ -0,0 +1,81 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// NOTE: This test will mutate the default ingresscontroller
+func TestTunableMaxConnections(t *testing.T) {
+	t.Logf("waiting for ingresscontroller %v", defaultName)
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, defaultName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller deployment: %v", err)
+	}
+
+	if err := waitForDeploymentEnvVar(t, kclient, deployment, time.Minute, ingress.RouterMaxConnectionsEnvName, ""); err != nil {
+		t.Fatalf("expected router deployment to have %s unset: %v", ingress.RouterMaxConnectionsEnvName, err)
+	}
+
+	for _, testCase := range []struct {
+		description    string
+		maxConnections int32
+		expectedEnvVar string
+	}{
+		{"set maxconn 12345", 12345, "12345"},
+		{"set maxconn auto", -1, "auto"},
```

Comment:
Just curious, have you prepared the PR to support "-1" as auto in openshift-router? I know you are playing with some changes with maxConn reservations.

Reply:
When we specify -1 that becomes "auto". The template only populates a value if it is an integer (https://github.com/openshift/router/blob/master/images/router/haproxy/conf/haproxy-config.template#L48). Setting a non-integer (i..e, "auto") omits any value when the template generation is run.

Reply:
Ah that's right. I forgot if you omit a maxconn value, it does auto.